physics/PBL/SATMEDMF/satmedmfvdifq.meta

```diff
@@ -217,6 +217,29 @@
   type = real
   kind = kind_phys
   intent = in
+[usfco]
+  standard_name = ocn_current_zonal
+  long_name = ocn_current_zonal
```

Comment:
@grantfirl, I know at the FV3ATM level, usfco and vsfco are earth relevant (zonal, meridional). But I am not sure its convention at the ccpp-physics level (@binli2337 could you please help to double check?). Meanwhile, a quick question in case you know on top of your head, are u10m and v10m inside ccpp-physics earth relevant (zonal, meridional) or grid relevant? I assume u1 and v1 uses the same convention. Thanks!

Reply:
@grantfirl @BinLiu-NOAA @junwang-noaa The ocean current fields, usfco and vsfco are truly zonal and meridional. From my understanding, the surface winds, u1 and v1 in sfc_diff.f are not truly zonal and meridional. Since we need to compute u1-usfco and v1-vsfco, we need to do a interpolation for ocean currents to convert usfco and vsfco from zonal and meridional to x and y in the native model grids. We need instructions on how to do the conversion.

Reply:
@BinLiu-NOAA and @binli2337 It is also my understanding that all u and v values within physics are grid-relative, not Earth-relative. I think after dynamics, u and v (and all other similar variables) are calculated on the A-grid before being passed to physics. The same thing should be done to usfco and vsfco, in my opinion, before entering physics. I'd check with the GFDL/FV3 folks for more info on how to do this efficiently. Once done, the standard names should change to using the same standard name conventions (x_ocean_current and y_ocean_current).

Reply:
@binli2337, @junwang-noaa and I had a conversation this morning, it seems to us the wind components used inside ccpp-physics are already earth-relevant (zonal/meridional). In this case, the expected usfco/vsfco (ocean surface current) should also be earth-relevant (which I think is already the case). @grantfirl or @Qingfu-Liu or others, appreciate if someone can help to confirm this is the case. Thanks!

Reply:
I'm not an expert in the dycore, but my reading of https://www.gfdl.noaa.gov/wp-content/uploads/2018/06/Harris-training-June2018-FV3-structure-1.pdf is that u and v are grid-relative (they use the term "local coordinate"). They explicitly say that a transform needs to happen to obtain zonal and meridional winds. So, if usfco/vsfco are truly zonal/meridional, I still think that they need to be converted to the "local coordinate". I would talk to Lucas Harris or someone else who is an expert to verify.

Reply:
@grantfirl, @binli2337, and @junwang-noaa, I am trying to dig into the FV3dycore code, it looks to me that the ua/va (assuming they are used in ccpp-physics) were rotated from model grid (cubed sphere grid) to latlon grid relative in this code section:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/65301dd7c6b29c755a1f31b1a8e94562d3bd920c/model/fv_update_phys.F90#L750
```
#ifdef GFS_PHYS
    call cubed_to_latlon(u, v, ua, va, gridstruct, &
         npx, npy, npz, 1, gridstruct%grid_type, domain, gridstruct%bounded_domain, flagstruct%c2l_ord, bd)
#endif
```

But , again, my understanding might be wrong. Thanks!

Reply:
Here is the double check that from the connection between dycore and physics:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/65301dd7c6b29c755a1f31b1a8e94562d3bd920c/driver/fvGFS/atmosphere.F90#L2114
```
         IPD_Data(nb)%Statein%ugrs(ix,k) = _DBL_(_RL_(Atm(mygrid)%ua(i,j,k1)))
         IPD_Data(nb)%Statein%vgrs(ix,k) = _DBL_(_RL_(Atm(mygrid)%va(i,j,k1)))
```
the ugrs/vgrs used in ccpp-physics are from ua/va, which should be earth-relative (zonal/meridional). Thanks!

Reply:
Are you sure that cubed_to_latlon means what you think it means? There is no in-line documentation.

Reply:
For consistency, however, regardless of the orientation, I'd still change the standard names to using x and y. You can put zonal and meridional in the long name. We adopted x_wind and y_wind because they are more general so that physics can be used in models that perhaps don't use Earth-relative, true zonal/meridional winds/currents.

Reply:
> Are you sure that cubed_to_latlon means what you think it means? There is no in-line documentation.

The actual code doing the rotation is here for ua/va:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/65301dd7c6b29c755a1f31b1a8e94562d3bd920c/model/fv_grid_utils.F90#L2638C1-L2645C8
```
 !Transform local a-grid winds into latitude-longitude coordinates
     do j=js,je
        do i=is,ie
           ua(i,j,k) = gridstruct%a11(i,j)*utmp(i,j) + gridstruct%a12(i,j)*vtmp(i,j)
           va(i,j,k) = gridstruct%a21(i,j)*utmp(i,j) + gridstruct%a22(i,j)*vtmp(i,j)
        enddo
     enddo
```
or here:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/65301dd7c6b29c755a1f31b1a8e94562d3bd920c/model/fv_grid_utils.F90#L2722C1-L2724C57
```
! Cubed (cell center co-variant winds) to lat-lon:
             ua(i,j,k) = a11(i,j)*u1(i) + a12(i,j)*v1(i)
             va(i,j,k) = a21(i,j)*u1(i) + a22(i,j)*v1(i)
```

But if someone from the FV3ATM or the FV3dycore or ccpp-physics side can confirm that will be better. Thanks!

Reply:
> standard name conventions (x_ocean_current and y_ocean_current).

@grantfirl, using the standard name of x_ocean_current and y_ocean_current (or x_current/y_current or x_current_ocean/y_current_ocean) sounds good/reasonable to me. As for the long_names, how about  "zonal current at ocean surface" and "meridional current at ocean surface"? Please feel free to provide suggestions or let us know your preferences. Thanks!

Reply:
@BinLiu-NOAA They sound fine to me. I couldn't find a CF variable that was similar, so you're free to choose. I have a small preference for including the word "ocean" in the name to be super clear, but other than that, they're all fine. The long names are fine too. They're mainly for documentation. Thanks for checking on the zonal/meridional thing. I had been referring to the documentation for a long time and knew that an A-grid transformation took place, but not a transform to zonal/meridional, so I appreciate learning this!

Reply:
Thanks for all the discussions and suggestions, @grantfirl! @binli2337, with that, let's use x_ocean_current and y_ocean_current for standard_names and "zonal current at ocean surface" and "meridional current at ocean surface" for long_names. Appreciate your help to make the changes in both the ccpp-physics and FV3ATM levels. Thanks!